CMakeLists.txt

```diff
@@ -2,11 +2,40 @@ cmake_minimum_required(VERSION 3.4)
 
 project(ray)
 
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules")
+
+# This ensures that things like gnu++11 get passed correctly
+set(CMAKE_CXX_STANDARD 11)
+
+# We require a C++11 compliant compiler
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+option(RAY_BUILD_STATIC
+  "Build the libarrow static libraries"
+  ON)
+
+option(RAY_BUILD_SHARED
+  "Build the libarrow shared libraries"
+  ON)
+
+option(RAY_BUILD_TESTS
```

Comment:
is this used anywhere?

Reply:
It is used in ThirdpartyToolchain.cmake now (I added it)